test/hts-precompile/utils.js

```diff
@@ -668,6 +671,158 @@ class Utils {
     const txResponse = await signTx.execute(signerClient)
     await txResponse.getReceipt(signerClient)
   }
+
+  /**
+   * @dev Constructs a key conforming to the IHederaTokenService.TokenKey type
+   *
+   * @param keyType ADMIN | KYC | FREEZE | WIPE | SUPPLY | FEE | PAUSE
+   *                See https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L128
+   *                for more information
+   *
+   * @param keyValyeType INHERIT_ACCOUNT_KEY | CONTRACT_ID | ED25519 | SECP256K1 | DELEGETABLE_CONTRACT_ID
+   *
+   * @param key bytes value, public address of an account, or boolean
+   *            See https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L92
+   *                     for more information
+   */
+  static constructIHederaTokenKey(keyType, keyValueType, key) {
```

Comment:
Much better.
Nice improvement :) 
Should scrap `prepareTokenKeysArray()` and let callers combine this as needed